src/legacyConditions.ts

```diff
@@ -0,0 +1,172 @@
+import type {LiquidContext} from './types';
```

Comment:
If this is a temporary backport for a short period (e.g., urgent fix or migration), we can merge without tests.
However, if legacyConditions is intended to be supported for a while, we should add at least basic testcases.
Although, I think it would be better to spend time migrating to the current liquid implementation rather than maintaining legacy code.